### PR TITLE
Custom deserializer

### LINKED ISSUE #4: Add ability for consumer defined deserializer
Add the ability to override the default JSON deserializer allowing the case of a special JSON structure. This will require the consumer to convert this to our model but allows any JSON structure to be used.

This should also included an updated README section at the bottom informing the consumer of the ability to provide a custom deserializer and thus custom JSON.

### PR DESCRIPTION
**Pull request by btkelly**
Added the ability to provide a custom deserializer to accommodate any JSON structure a consumer might have. I have also updated the README to demonstrate how to do this. Resolves #4

### CODE REVIEW

File: gandalf/src/main/java/io/github/btkelly/gandalf/models/Bootstrap.java [hunk 1]
```diff
@@ -83,4 +85,54 @@ public Bootstrap createFromParcel(Parcel source) {
             return new Bootstrap[size];
         }
     };
+
+    public static class Builder {
```
Comment (jsibbold) [inline gandalf/src/main/java/io/github/btkelly/gandalf/models/Bootstrap.java]: for the builder, how about instead doing
```
builder.setAlert(message, blocking)
           .setOptionalUpdate(message, version)
           .setRequiredUpdate(message, version);
```
might feel a bit less cumbersome, and also make you less likely to miss a value that you should have.

Comment (stkent) [inline gandalf/src/main/java/io/github/btkelly/gandalf/models/Bootstrap.java]: +1 this.

Comment (btkelly) [inline gandalf/src/main/java/io/github/btkelly/gandalf/models/Bootstrap.java]: Yeah not bad, I'm on it

**Comment by jsibbold** [comment]
Approved

**Comment by stkent** [comment]
Approved

**Pull request merged**